pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1110,25 +1100,22 @@ func (idxcPlayBookContext *IdxcPlayBookContext) runPlayBook() error {
 // afwSchedulerEntry Starts the scheduler Pipeline with the required phases
 func afwSchedulerEntry(client splcommon.ControllerClient, cr splcommon.MetaObject, appDeployContext *enterpriseApi.AppDeploymentContext, appFrameworkConfig *enterpriseApi.AppFrameworkSpec) error {
 	scopedLog := log.WithName("afwSchedulerEntry").WithValues("name", cr.GetName(), "namespace", cr.GetNamespace())
-	afwPipeline = initAppInstallPipeline(appDeployContext)
+	afwPipeline := initAppInstallPipeline(appDeployContext)
 
 	var clusterScopedApps []*enterpriseApi.AppDeploymentInfo
 
 	afwEntryTime := time.Now().Unix()
+	var err error
 
-	// ToDo: sgontla: for now, just make the UT happy, until we get the glue logic
-	// check if this needs to be pure singleton for the entire reconcile span, considering CSPL-1169. CC: @Gaurav
-	// Finally delete the pipeline
-	defer func() {
-		afwPipeline = nil
-	}()
-
-	// get the current available disk space for downloading apps on operator pod
-	availableDiskSpace, err := getAvailableDiskSpace()
-	if err != nil {
-		return err
+	// Check if the persistant storage is configured
+	if !isPersistantVolConfigured() {
+		scopedLog.Error(nil, "persistant volume required for the App framework, but not provisioned")
+		operatorResourceTracker = &globalResourceTracker{}
+		err = initStorageTracker()
+		if err != nil {
+			scopedLog.Error(err, "unable to initialize storage tracker")
+		}
```

Comment:
Sorry for the basic question but aren't we doing this thing already in the `init` function in util.go?